### PR TITLE
Update provider, add gcp_filestore_csi_driver_config block, make variable optional

### PR DESCRIPTION
**Pull request by soerenmartius**
- Change unit-minimal to deploy a routes based cluster
- Turn `var.location` to be optional
- Change `var.network` and `var.subnetwork` to be optional variables
- Change minimum required version of google provider to `v4.10.0`

### CODE REVIEW

**Review by mariux — APPROVED** [review]
lgtm

File: test/unit-minimal/main.tf [hunk 1]
```diff
@@ -87,19 +95,11 @@ module "test" {
...
-  networking_mode        = "VPC_NATIVE"
-  master_ipv4_cidr_block = "10.4.96.0/28"
-  ip_allocation_policy = {
-    cluster_secondary_range_name  = local.subnet.secondary_ip_ranges.pods.name
-    services_secondary_range_name = local.subnet.secondary_ip_ranges.services.name
-  }
+  name = "gke-unit-minimal"
+
+  # deploy a public cluster with routes based networking mode
+  networking_mode      = "ROUTES"
```
Comment (mariux) [inline test/unit-minimal/main.tf:101]: please deploy the recommended vpc native cluster

**Review by mariux — CHANGES_REQUESTED** [review]
lgtm

File: test/unit-minimal/main.tf (hunk 1 above)
Comment (soerenmartius) [reply test/unit-minimal/main.tf:101]: we already do this in the unit-complete test

Comment (soerenmartius) [reply test/unit-minimal/main.tf:101]: what's the reasoning behind this?

Comment (soerenmartius) [reply test/unit-minimal/main.tf]: resolved

**Review by kassianh — APPROVED** [review]
lgtm, just need to fix the missing variable in the test

**Review by kassianh — APPROVED** [review]
lgtm

**Pull request merged**